lib/srv/discovery/discovery_test.go

```diff
@@ -991,26 +990,49 @@ func TestDiscoveryServer(t *testing.T) {
 						require.Equal(t, expectedValue, got.IntegrationDiscoveredResources[expectedKey])
 					}
 					return true
-				}, 500*time.Millisecond, 50*time.Millisecond)
+				}, 1*time.Second, 50*time.Millisecond)
 			}
 			if tc.userTasksDiscoverCheck != nil {
-				var allUserTasks []*usertasksv1.UserTask
-				var nextToken string
-				for {
-					var userTasks []*usertasksv1.UserTask
-					userTasks, nextToken, err = tlsServer.Auth().UserTasks.ListUserTasks(context.Background(), 0, "", &usertasksv1.ListUserTasksFilters{})
-					require.NoError(t, err)
-					allUserTasks = append(allUserTasks, userTasks...)
-					if nextToken == "" {
-						break
-					}
-				}
-				tc.userTasksDiscoverCheck(t, allUserTasks)
+				tc.userTasksDiscoverCheck(t, tlsServer.Auth().UserTasks)
 			}
 		})
 	}
 }
 
+func fetchAllUserTasks(t *testing.T, userTasksClt services.UserTasks, minUserTasks, minUserTaskResources int) []*usertasksv1.UserTask {
+	var existingTasks []*usertasksv1.UserTask
+	require.Eventually(t, func() bool {
+		var allTasks []*usertasksv1.UserTask
+		var nextToken string
+		for {
+			var userTasks []*usertasksv1.UserTask
+			userTasks, nextTokenResp, err := userTasksClt.ListUserTasks(context.Background(), 0, nextToken, &usertasksv1.ListUserTasksFilters{})
+			require.NoError(t, err)
+			allTasks = append(allTasks, userTasks...)
+			if nextTokenResp == "" {
+				break
+			}
+			nextToken = nextTokenResp
+		}
+		existingTasks = allTasks
+
+		if len(allTasks) < minUserTasks {
+			return false
+		}
+
+		gotResources := 0
+		for _, task := range allTasks {
+			gotResources += len(task.GetSpec().GetDiscoverEc2().GetInstances())
+			gotResources += len(task.GetSpec().GetDiscoverEks().GetClusters())
+			gotResources += len(task.GetSpec().GetDiscoverRds().GetDatabases())
+		}
+
+		return gotResources >= minUserTaskResources
+	}, 5*time.Second, 1*time.Second)
```

Comment:
How long are we supposed to wait for the task? One second ticks are sort of a lot.

Reply:
Yeah, you are right.
Decreased it to be similar to other tick intervals: 50ms